### DISCUSSION TITLE
Caisson

### DISCUSSION THREAD

**Discussion by salieflewis** [public-assembly/.github#4] [Announcements]
Caisson is a frontend development template designed to grow alongside the Public Assembly engineering community.

If you have suggestions for what this template needs, or want to contribute to any aspect of itss development, branding, or design, say hello in this thread or open a pull request!

@neatonk @dblodorn @E-Rick @0xTranqui @mmsaki @thekidnamedkd

**Comment by 0xTranqui** [comment]
thank u for doing this. i have a lot of ideas in terms of common patterns/snippets/etc I use over and over again that would be awesome to standardize into this I think. to give you an example of what im talking bout, see the code snippet below:

```
// state for loading txns
const svgLoader = () => {
    return (
        <div className="flex flex-row justify-center items-center w-full ">
            <img
            width="20px"
            src="/SVG-Loaders/svg-loaders/tail-spin.svg"
            />
        </div>
    )
}

const curateSpinner = isLoading || mintWaitLoading
    ? svgLoader()
    : "curate"     
    
<button
  className=""
  onClick={()=>write()}
  disabled={isLoading || mintWaitLoading ? true : false}
>
  {curateSpinner}
</button>      
```

this pattern combined with this pretty [classic svg loader library](https://samherbert.net/svg-loaders/) + wagmi loading state returns from their useContractWrite + useWaitForTransaction hooks  is something I've been using ever since the Felt gardens project to handle loading states for onchain transactions. I think generalizing this could be really nice – lmk if this is the type of stuff you were thinking for this repo

**Reply by salieflewis** [reply]
I know we spoke about this in person, but still wanted to write a reply obviously for the sake of everyone else. Personally I think that the pattern you're describing lands more in the design system territory than say this template.

**Comment by thekidnamedkd** [comment]
What do you think of Rome? @salieflewis

**Reply by salieflewis** [reply]
This was actually the first project I tried using it with. I have yet to even fork this template myself, so I think some adjustments will be made after doing that and experimenting with some of the scripts.

Would love some of your feedback if you've used it or are interested in trying it out via Caisson!
